docs/pallas/gpu/reference.md

```diff
@@ -0,0 +1,144 @@
+# Writing Mosaic GPU kernels with Pallas
+
+This page is a reference for the most important features of the Pallas:MGPU backend.
```

Comment:
Super-nit: should we be saying "Pallas Mosaic GPU" here to be consistent with the header?